adapters/tlsio_mbedtls.c

```diff
@@ -677,10 +677,19 @@ int tlsio_mbedtls_send(CONCRETE_IO_HANDLE tls_io, const void *buffer, size_t siz
             tls_io_instance->send_complete_info.on_send_complete = on_send_complete;
             tls_io_instance->send_complete_info.on_send_complete_callback_context = callback_context;
             tls_io_instance->send_complete_info.send_complete_count = 0;
-            int res = mbedtls_ssl_write(&tls_io_instance->ssl, buffer, size);
-            if (res != (int)size)
+
+            int bytes_sent = 0;
+            int res = 0;
+
+            do{
+                res = mbedtls_ssl_write(&tls_io_instance->ssl, buffer+bytes_sent, size-bytes_sent);
+                bytes_sent += res;
```

Comment:
This will cause on_send_complete to be fired  on first chunk of data sent successfully ... but this should be opposite on_send_complete should fire when all the data push has been complete.

Check line  347 of this file.

Reply:
Did not see this problem... 

I added a `on_partial_send_complete` function with counters to call the actual `on_send_complete` function only when all chunk are sent. What do you think?

Reply:
I have create a PR for this fix
https://github.com/Azure/azure-c-shared-utility/pull/421

Let me know if you still see this issue after applying my commit.